### PR TITLE
Configuration for preventing manual/user instantiation in production

### PR DESCRIPTION
**Pull request by martinothamar**
## Description
We need configuration to block "manual instantiation" by users, for applications where service owner
create and prefill instances and then let users fill in the rest.

Related PR: 
- https://github.com/Altinn/altinn-storage/pull/426

## Related Issue(s)
- https://github.com/Altinn/app-lib-dotnet/issues/361

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.App.Api/Controllers/InstancesController.cs [hunk 1]
```diff
@@ -445,6 +452,24 @@ IOrganizationClient orgClient
             );
         }
 
+        // TODO: is production enough? I see that tt02 is Staging, unsure about other environments
```
Comment (RonnyB71) [inline src/Altinn.App.Api/Controllers/InstancesController.cs]: I think production should be enough.

**Review by RonnyB71 — APPROVED** [review]
Looks good, only need it to be green:)

File: src/Altinn.App.Core/Models/ApplicationMetadata.cs [hunk 2]
```diff
@@ -70,4 +70,7 @@ public new string Id
     /// </summary>
     [System.Text.Json.Serialization.JsonExtensionData]
     public Dictionary<string, object>? UnmappedProperties { get; set; }
+
+    /// ! TEMP
```
Comment (danielskovli) [inline src/Altinn.App.Core/Models/ApplicationMetadata.cs]: Write some clever words here?

**Review by danielskovli — APPROVED** [review]
LGTM 🙌

**Review by RonnyB71 — APPROVED** [review]
Looks good:) The InstancesController class is bfu, bit that is for another time to solve.

**Review by bjorntore — APPROVED** [review]

**Pull request merged**